admin/server/projects.go

```diff
@@ -174,8 +175,13 @@ func (s *Server) SearchProjectNames(ctx context.Context, req *adminv1.SearchProj
 		return nil, err
 	}
 	pageSize := validPageSize(req.PageSize)
+	var projectNames []string
 
-	projectNames, err := s.admin.DB.FindProjectPathsByPattern(ctx, req.NamePattern, token.Val, pageSize)
+	if req.Tags != nil && len(req.Tags) > 0 {
+		projectNames, err = s.admin.DB.FindProjectPathsByPatternAndTags(ctx, req.NamePattern, token.Val, req.Tags, pageSize)
```

Comment:
nit: move the newline up one line (so declaration of `projectNames` goes in same group as assignment)